### PR TITLE
Portal dashboard design

### PR DESCRIPTION
**Pull request by BintLopez**
image

### CODE REVIEW

**Comment by BintLopez** [comment]
@nkeyes -- wip but looping ya in

**Comment by BintLopez** [comment]
@nkeyes fyi

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -9,39 +11,53 @@ def show
...
+        else
+          params.require(:user).require(expected).permit(permitted_keys)
+        end
+      end
+    end.compact.deep_symbolize_keys
   end
 
-  def address_present?
-    params["user"]["address"].values.any? {|v| !v.blank? }
+  def params_sans_empty_strings
```
Comment (nkeyes) [inline app/controllers/users_controller.rb]: this could/should be moved to a more generic place, as it has general purpose usefulness.

Comment (nkeyes) [inline app/controllers/users_controller.rb]: also, this only does the top level, us that okay?

File: app/models/person.rb [hunk 2]
```diff
@@ -10,15 +10,16 @@ def age
...
 
   def phone_number
-    contacts.current_phone_number
+    # TODO don't just go off last here
+    phone_numbers.last
   end
 
   def email
-    contacts.current_email
+    user.try(:email) || emails.last
```
Comment (nkeyes) [inline app/models/person.rb]: if you're on ruby 2.3.0+:
http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

File: app/models/person.rb [hunk 3]
```diff
@@ -10,15 +10,16 @@ def age
...
   end
 
   def address
-    addresses.active_current
+    addresses.last
```
Comment (nkeyes) [inline app/models/person.rb]: postgres doesn't have guaranteed ordering.

File: app/models/phone_number.rb [hunk 4]
```diff
@@ -1,3 +1,18 @@
 class PhoneNumber < ActiveRecord::Base
   belongs_to :callable, polymorphic: true
+  validate :validate_number
+
+  VALID_NUMBER_LENGTH = 10.freeze
```
Comment (nkeyes) [inline app/models/phone_number.rb]: probably an edge case, but you want to account for 1 as the US country code

File: app/models/user.rb [hunk 5]
```diff
@@ -34,6 +35,17 @@ def self.find_for_google_oauth2(auth)
...
 
+  def full_name
+    person.try(:full_name)
+  end
+
+  def address
+    person.try(:address)
+  end
+
+  def board_member?
```
Comment (nkeyes) [inline app/models/user.rb]: this is made redundant by lines 50-52

File: app/presenters/organizer_presenter.rb [hunk 6]
```diff
@@ -17,7 +16,7 @@ def data
 
   def organizer_data
     {
-      board_member: organizer.board_member,
+      role: user.board_member,
```
Comment (nkeyes) [inline app/presenters/organizer_presenter.rb]: is this supposed to be `user.role` or user.board_member?`?

**Pull request merged**